.github/assets/check_wasm.sh

```diff
@@ -1,29 +1,111 @@
 #!/usr/bin/env bash
 set +e  # Disable immediate exit on error
 
-# Array of crates 
-wasm_crates=(
-  # The following were confirmed not working in the past, but could be enabled if issues have been resolved
-  # reth-consensus
-  # reth-db
-  # reth-evm
-  # reth-evm-ethereum
-  # The following are confirmed working
-  reth-codecs
-  reth-errors
-  reth-ethereum-forks
-  reth-network-peers
-  reth-primitives
-  reth-primitives-traits
-  reth-revm
+# Array of crates to compile
+crates=($(cargo metadata --format-version=1 --no-deps | jq -r '.packages[].name' | grep '^reth' | sort))
+# Array of crates to exclude
+exclude_crates=(
+  # The following are not working yet, but known to be fixable
+  reth-consensus
+  reth-exex-types # https://github.com/paradigmxyz/reth/issues/9946
+  reth-evm
+  # The following require investigation if they can be fixed
+  reth-auto-seal-consensus
+  reth-basic-payload-builder
+  reth-beacon-consensus
+  reth-bench
+  reth-blockchain-tree
+  reth-chain-state
+  reth-cli
+  reth-cli-commands
+  reth-cli-runner
+  reth-consensus-debug-client
+  reth-db-common
+  reth-discv4
+  reth-discv5
+  reth-dns-discovery
+  reth-downloaders
+  reth-e2e-test-utils
+  reth-engine-primitives
+  reth-engine-tree
+  reth-engine-util
+  reth-eth-wire
+  reth-ethereum-cli
+  reth-ethereum-engine
+  reth-ethereum-engine-primitives
+  reth-ethereum-payload-builder
+  reth-etl
+  reth-evm-ethereum
+  reth-evm-optimism
+  reth-execution-errors
+  reth-exex
+  reth-exex-test-utils
+  reth-ipc
+  reth-net-nat
+  reth-network
+  reth-node-api
+  reth-node-builder
+  reth-node-core
+  reth-node-ethereum
+  reth-node-events
+  reth-node-metrics
+  reth-node-optimism
+  reth-optimism-cli
+  reth-optimism-payload-builder
+  reth-optimism-rpc
+  reth-payload-builder
+  reth-payload-primitives
+  reth-rpc
+  reth-rpc-api
+  reth-rpc-api-testing-util
+  reth-rpc-builder
+  reth-rpc-engine-api
+  reth-rpc-eth-api
+  reth-rpc-eth-types
+  reth-rpc-layer
+  reth-rpc-types
+  reth-stages
+  reth-storage-errors
+  # The following are not supposed to be working
+  reth # all of the crates below 
+  reth-db # mdbx
+  reth-libmdbx # mdbx
+  reth-mdbx-sys # mdbx
+  reth-nippy-jar # sucds
+  reth-provider # reth-db, reth-nippy-jar
+  reth-prune # reth-db
+  reth-stages-api # reth-provider, reth-prune
+  reth-static-file # reth-nippy-jar 
+  reth-transaction-pool # c-kzg
+  reth-trie-db # reth-db 
+  reth-trie-parallel # reth-db
```

Comment:
@mattsse can you confirm these aren't supposed to work? I'm not sure about some of them, e.g. feels like `reth-transaction-pool` and `reth-trie-parallel` should be decoupled from `c-kzg` and db accordingly?

Reply:
yeah but not a big deal imo